### ISSUE TITLE
Layer hierarchy potentially overly flat

### ISSUE THREAD

**Issue by realmarcin** [berkeleybop/artificial-intelligence-ontology#117]
Consider the context of an AI ontology, which of these DNN Layers could be grouped into categories?

GPT4o
####
Normalization Layers
Batch Normalization Layers:

BatchNorm1D Layer
BatchNorm2D Layer
BatchNorm3D Layer
SyncBatchNorm Layer
LazyBatchNorm1D Layer
LazyBatchNorm2D Layer
LazyBatchNorm3D Layer
BatchNormalization Layer
Instance Normalization Layers:

InstanceNorm1D Layer
InstanceNorm2D Layer
InstanceNorm3D Layer
LazyInstanceNorm1D Layer
LazyInstanceNorm2D Layer
LazyInstanceNorm3D Layer
Other Normalization Layers:

GroupNorm Layer
LayerNorm Layer
LayerNormalization Layer
LocalResponseNorm Layer
UnitNormalization Layer
Normalization Layer
Convolutional Layers
Standard Convolutional Layers:

Convolution1D Layer
Convolution2D Layer
Convolution3D Layer
Transposed Convolutional Layers:

Convolution1DTranspose Layer
Convolution2DTranspose Layer
Convolution3DTranspose Layer
Depthwise Convolutional Layers:

DepthwiseConv1D Layer
DepthwiseConv2D Layer
Separable Convolutional Layers:

SeparableConvolution1D Layer
SeparableConvolution2D Layer
Convolutional LSTM Layers:

ConvLSTM1D Layer
ConvLSTM2D Layer
ConvLSTM3D Layer
Other Convolutional Layers:

LocallyConnected1D Layer
LocallyConnected2D Layer
Locally-connected Layer
Pooling Layers
Average Pooling Layers:

AvgPool1D Layer
AvgPool2D Layer
AvgPool3D Layer
AveragePooling1D Layer
AveragePooling2D Layer
AveragePooling3D Layer
GlobalAveragePooling1D Layer
GlobalAveragePooling2D Layer
GlobalAveragePooling3D Layer
Max Pooling Layers:

MaxPooling1D Layer
MaxPooling2D Layer
MaxPooling3D Layer
GlobalMaxPooling1D Layer
GlobalMaxPooling2D Layer
GlobalMaxPooling3D Layer
Adaptive Pooling Layers:

AdaptiveAvgPool1D Layer
AdaptiveMaxPool1D Layer
AdaptiveAvgPool2D Layer
AdaptiveMaxPool2D Layer
AdaptiveAvgPool3D Layer
AdaptiveMaxPool3D Layer
Other Pooling Layers:

FractionalMaxPool2D Layer
FractionalMaxPool3D Layer
MaxUnpool1D Layer
MaxUnpool2D Layer
MaxUnpool3D Layer
LPPool1D Layer
LPPool2D Layer
RNN Layers
Standard RNN Layers:

SimpleRNN Layer
GRU Layer
LSTM Layer
RNN Layer
RNN Cell Layers:

GRUCell Layer
SimpleRNNCell Layer
LSTMCell Layer
AbstractRNNCell
StackedRNNCells Layer
Bidirectional and Time-Distributed Layers:

Bidirectional Layer
TimeDistributed Layer
Attention Layers
Attention Mechanisms:
Attention Layer
MultiHeadAttention Layer
AdditiveAttention Layer
Preprocessing Layers
Categorical Features Preprocessing:

CategoryEncoding Layer
IntegerLookup Layer
StringLookup Layer
Hashing Layer
Numerical Features Preprocessing:

Discretization Layer
Text Preprocessing:

TextPreprocessing Layer
TextVectorization Layer
Image Preprocessing and Augmentation:

ImagePreprocessing Layer
ImageAugmentation Layer
CenterCrop Layer
RandomBrightness Layer
RandomContrast Layer
RandomCrop Layer
RandomFlip Layer
RandomRotation Layer
RandomTranslation Layer
RandomHeight Layer
RandomWidth Layer
RandomZoom Layer
Rescaling Layer
Resizing Layer
Dense and Hidden Layers
Dense Layers:

Dense Layer
DenseFeatures Layer
Hidden Layers:

Hidden Layer
Probabilistic Hidden Layer
Spiking Hidden Layer
Activation Layers
Activation Functions:
Activation Layer
PReLU Layer
ELU Layer
LeakyReLU Layer
ReLU Layer
Softmax Layer
ThresholdedReLU Layer
Regularization Layers
Dropout and Noise Layers:
Dropout Layer
GaussianDropout Layer
SpatialDropout1D Layer
SpatialDropout2D Layer
SpatialDropout3D Layer
GaussianNoise Layer
AlphaDropout Layer
NoiseDense Layer
ActivityRegularization Layer
Input, Output, and Other Layers
Input Layers:

Input Layer
NoisyInput Layer
BackfedInput Layer
MatchedInputOutput Layer
Output Layers:

Output Layer
Wrapper Layers:

Wrapper Layer
Merging Layers:

Add Layer
Average Layer
Maximum Layer
Minimum Layer
Concatenate Layer
Multiply Layer
Subtract Layer
Dot Layer
Reshaping and Permutation Layers:

Reshape Layer
Permute Layer
Flatten Layer
RepeatVector Layer
Cropping and Padding Layers:

Cropping1D Layer
Cropping2D Layer
Cropping3D Layer
ZeroPadding1D Layer
ZeroPadding2D Layer
ZeroPadding3D Layer
Upsampling Layers:

UpSampling1D Layer
UpSampling2D Layer
UpSampling3D Layer
Other Layers:

Lambda Layer
Policy Layer
MemoryCell Layer
Weighted Layer
Layer Layer
InputSpec Layer
Masking Layer
Kernel Layer

**Comment by cmungall** [comment]
Looks like this is done?
